tests/modules/segmentation/test_processing_log.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+import h5py
+import datetime
+import contextlib
+import numpy as np
+
+from ophys_etl.types import ExtractROI
+from ophys_etl.modules.segmentation import processing_log
+
+
+@pytest.fixture
+def extract_roi_list():
+    nroi = 20
+    xs = np.linspace(start=10, stop=490, num=nroi, dtype=int)
+    ys = np.linspace(start=20, stop=470, num=nroi, dtype=int)
+    widths = ([10] * (nroi - 10) + [12] * 10)
+    heights = ([8] * (nroi - 7) + [11] * 7)
+    valids = ([True] * (nroi - 4) + [False] * 4)
+
+    rng = np.random.default_rng(3342)
+    roi_list = []
+    for i in range(nroi):
+        mask = [row.tolist()
+                for row in rng.integers(low=0,
+                                        high=2,
+                                        size=(heights[i], widths[i])
+                                        ).astype(bool)]
+        roi_list.append(
+                ExtractROI(
+                    id=int((i + 1)),
+                    x=int(xs[i]),
+                    y=int(ys[i]),
+                    width=int(widths[i]),
+                    height=int(heights[i]),
+                    mask=mask,
+                    valid=valids[i]))
+    return roi_list
+
+
+def test_timestamp(tmpdir):
+    h5path = tmpdir / "test.h5"
+    with h5py.File(h5path, "w") as f:
+        group = f.create_group("timestamp_test")
+        processing_log.timestamp_group(group)
+        assert "group_creation_time" in group
+        dstr = group["group_creation_time"][()].decode("utf-8")
+
+    assert isinstance(datetime.datetime.fromisoformat(dstr),
+                      datetime.datetime)
+
+
+@pytest.mark.parametrize(
+        "read_only, context",
+        [
+            (True, pytest.warns(
+                UserWarning, match=r".*can not be invoked.*")),
+            (False, contextlib.nullcontext())
+            ])
+def test_read_only(tmpdir, extract_roi_list, read_only, context):
+    """tests that write methods emit a warning and don't write
+    when read_only=False
+    """
+    h5path = tmpdir / "test.h5"
+    plog = processing_log.SegmentationProcessingLog(path=h5path,
+                                                    read_only=read_only)
+    assert plog.read_only == read_only
+
+    # log_detection should be read only when asked
```

Comment:
Sorry for being pedantic, but do you mean "log_detection should be read-only when asked"? (note the hyphen)

The first few times I read these comments, I interpreted "read" as the past tense of "to read", which didn't make sense to me. This might come up when someone else has to debug these tests.

Reply:
clarified these.